docs_app/content/tutorial/7-flattening-operations.md

```diff
@@ -0,0 +1,132 @@
+
+## Flattening Observables Manually
+
+In order to flatten an Observable, we must first have an "observable of observables", or in TypeScript `Observable<Observable<T>>`. The flattening process is complicated to do correctly with proper semantics, but if you were to try to do it manually, it might look like this:
+
+```ts
+/**
+ * An example flatten operator (DO NOT USE)
+ */
+function flattenAll<T>(source: Observable<Observable<T>>) {
+    // We take our Observable<Observable<T>> and return an Observable<T>
+    return new Observable<T>(subscriber => {
+        const subscription = new Subscription();
+
+        // Subscribe to the "outer" source
+        subscription.add(source.subscribe({
+            next(innerSource: Observable<T>) {
+                // Subscribe to each "inner source" to get their values
+                subscription.add(
+                    innerSource.subscribe({
+                        next(value: T) {
+                            // Push the values from the inner source to the consumer
+                            subscriber.next(value),
+                        }
+                        // ??? What to do here? See below
+                    })
+                )
+            },
+            // ???? What to od here? See below.
+        }));
+        return subscription;
+    })
+}
+```
+
+Now the above example is pretty simple, but it's really just to give you the basic idea of how we can flatten an `Observable<Observable<T>>`. There are some issues with it the example above:
+
+1. What happens when the inner subscription has an error?
+2. Do inner completion notifications matter?
+3. Do we really want to stop pushing values to the producer *just* because the outer producer has completed? What if the inner producers aren't complete yet? We won't get all of the values!
+
+These are all important matters, and RxJS resolves these questions with a set of guarantees around the primary flattening operations.
+
+## RxJS Flattening Operator Guarantees
+
+- **All errors are forwarded.** Any error on an outer or inner observable will be pushed do the consumer and stop the subscription.
+- **Source and inner sources must complete.** RxJS flattening operators will return observables whose [subscriptions](GL) will not [complete](GL) until:
+  1. The outer [subscription](GL) is [complete](GL).
+  2. All active inner [subscriptions](GL) have completed*.
+
+    (* there's some slight technical nuance here with the [`switchMap`](API) and [`switchAll`](API) operators.)
+
+## Merge (merge, mergeMap, mergeAll, mergeWith)
```

Comment:
Is `merge` really a flattening operator?

Reply:
"operation", yes. (In a creation function)